src/DataBase/EntryDatabase.cs

```diff
@@ -142,49 +142,30 @@ public async Task<bool> DeleteAsync(Expression<Func<IEntryModel, bool>> filter,
 
         public async Task<bool> DeleteListAsync(Expression<Func<IEntryModel, bool>> filter, EntryType entryType)
         {
-            if (entryType == EntryType.Income)
+            IEnumerable<IEntryModel> entriesToRemove = entryType switch
             {
-                var entriesToRemove = await Database.Incomes.Where(filter).Cast<IncomeModel>().ToListAsync();
-                Database.Incomes.RemoveRange(entriesToRemove);
-            }
-            else
-            {
-                var entriesToRemove = await Database.Expenses.Where(filter).Cast<ExpenseModel>().ToListAsync();
-                Database.Expenses.RemoveRange(entriesToRemove);
-            }
-
+                EntryType.Income => await Database.Incomes.Where(filter).Cast<IncomeModel>().ToListAsync(),
+                _ => await Database.Expenses.Where(filter).Cast<ExpenseModel>().ToListAsync()
+            };
+            Database.RemoveRange(entriesToRemove);
```

Comment:
How does this work, or source from where to read about?

Reply:
I think line is more clear when I removed the (as I understand) redundant cast

     await Database.Incomes.Where(filter).ToListAsync()


Source to read up on: https://docs.microsoft.com/en-us/dotnet/api/system.data.entity.queryableextensions.tolistasync?view=entity-framework-6.2.0